assets/app/styles/_spacers.less

```diff
@@ -0,0 +1,36 @@
+// Spacing utility classes
+// creates:
+//  .pad-sm, .pad-md, .pad-lg, etc
+//  .mar-sm, .mar-md, .mar-lg, etc
+//  .pad-left-sm, .mar-left-sm, etc
+//
+@sizes: none 0px e('!important'), sm 4px e(''), md 8px e(''), lg 12px e(''), xl 16px e(''), xxl 32px e('');
```

Comment:
Btw, bootstrap's version is based on `rem`, this isn't.  At present I'm not super accustomed to generating layout based on root em value. Thoughts on that welcome (but may be good not to adopt the BS naming convention then, as they probably will be quite different values).

As a pattern, `2,4,8,12,16,32` work well and are easy to generate balanced top,bottom, left, right rules.  To match `.gutter-top` we could go with `5,10,15,20,25,30`, but it would be interesting to know what px vals would actually look best.  


Updated the vars to look like this:

```less
@nil: none 0 e('!important');
// @auto 0 auto; is built into the fn below
@xs: xs 3px e('');      // 2px
@sm: sm 5px e('');      // 4px
@md: md 10px e('');     // 8px
@lg: lg 15px e('');     // 12px
@xl: xl 20px e('');     // 16px
@xxl: xxl 30px e('');   // 32px

@sizes: @nil, @xs, @sm, @md, @lg, @xl, @xxl;
@sides: top, right, bottom, left;
```

Which will generate this:
```css
.pad-auto-none {padding: 0 auto; }
.mar-auto-none {margin: 0 auto; }
.pad-none {padding: 0 0; }
.mar-none {margin: 0 0; }
.pad-top-none {padding-top: 0 !important; }
.mar-top-none {margin-top: 0 !important; }
.pad-right-none {padding-right: 0 !important; }
.mar-right-none {margin-right: 0 !important; }
.pad-bottom-none {padding-bottom: 0 !important; }
.mar-bottom-none {margin-bottom: 0 !important; }
.pad-left-none {padding-left: 0 !important; }
.mar-left-none {margin-left: 0 !important; }
.pad-auto-xs {padding: 0 auto; }
.mar-auto-xs {margin: 0 auto; }
.pad-xs {padding: 3px 4.5px; }
.mar-xs {margin: 3px 4.5px; }
.pad-top-xs {padding-top: 3px ; }
.mar-top-xs {margin-top: 3px ; }
.pad-right-xs {padding-right: 3px ; }
.mar-right-xs {margin-right: 3px ; }
.pad-bottom-xs {padding-bottom: 3px ; }
.mar-bottom-xs {margin-bottom: 3px ; }
.pad-left-xs {padding-left: 3px ; }
.mar-left-xs {margin-left: 3px ; }
.pad-auto-sm {padding: 0 auto; }
.mar-auto-sm {margin: 0 auto; }
.pad-sm {padding: 5px 7.5px; }
.mar-sm {margin: 5px 7.5px; }
.pad-top-sm {padding-top: 5px ; }
.mar-top-sm {margin-top: 5px ; }
.pad-right-sm {padding-right: 5px ; }
.mar-right-sm {margin-right: 5px ; }
.pad-bottom-sm {padding-bottom: 5px ; }
.mar-bottom-sm {margin-bottom: 5px ; }
.pad-left-sm {padding-left: 5px ; }
.mar-left-sm {margin-left: 5px ; }
.pad-auto-md {padding: 0 auto; }
.mar-auto-md {margin: 0 auto; }
.pad-md {padding: 10px 15px; }
.mar-md {margin: 10px 15px; }
.pad-top-md {padding-top: 10px ; }
.mar-top-md {margin-top: 10px ; }
.pad-right-md {padding-right: 10px ; }
.mar-right-md {margin-right: 10px ; }
.pad-bottom-md {padding-bottom: 10px ; }
.mar-bottom-md {margin-bottom: 10px ; }
.pad-left-md {padding-left: 10px ; }
.mar-left-md {margin-left: 10px ; }
.pad-auto-lg {padding: 0 auto; }
.mar-auto-lg {margin: 0 auto; }
.pad-lg {padding: 15px 22.5px; }
.mar-lg {margin: 15px 22.5px; }
.pad-top-lg {padding-top: 15px ; }
.mar-top-lg {margin-top: 15px ; }
.pad-right-lg {padding-right: 15px ; }
.mar-right-lg {margin-right: 15px ; }
.pad-bottom-lg {padding-bottom: 15px ; }
.mar-bottom-lg {margin-bottom: 15px ; }
.pad-left-lg {padding-left: 15px ; }
.mar-left-lg {margin-left: 15px ; }
.pad-auto-xl {padding: 0 auto; }
.mar-auto-xl {margin: 0 auto; }
.pad-xl {padding: 20px 30px; }
.mar-xl {margin: 20px 30px; }
.pad-top-xl {padding-top: 20px ; }
.mar-top-xl {margin-top: 20px ; }
.pad-right-xl {padding-right: 20px ; }
.mar-right-xl {margin-right: 20px ; }
.pad-bottom-xl {padding-bottom: 20px ; }
.mar-bottom-xl {margin-bottom: 20px ; }
.pad-left-xl {padding-left: 20px ; }
.mar-left-xl {margin-left: 20px ; }
.pad-auto-xxl {padding: 0 auto; }
.mar-auto-xxl {margin: 0 auto; }
.pad-xxl {padding: 30px 45px; }
.mar-xxl {margin: 30px 45px; }
.pad-top-xxl {padding-top: 30px ; }
.mar-top-xxl {margin-top: 30px ; }
.pad-right-xxl {padding-right: 30px ; }
.mar-right-xxl {margin-right: 30px ; }
.pad-bottom-xxl {padding-bottom: 30px ; }
.mar-bottom-xxl {margin-bottom: 30px ; }
.pad-left-xxl {padding-left: 30px ; }
.mar-left-xxl {margin-left: 30px ; }
```